src/Agent.Worker/JobRunner.cs

```diff
@@ -34,8 +36,14 @@ public sealed class JobRunner : AgentService, IJobRunner
             }
 
             // Setup the job server and job server queue.
+            var vssConnection = ApiUtil.GetVssConnection(message);
+
             var jobServer = HostContext.GetService<IJobServer>();
-            await jobServer.ConnectAsync(ApiUtil.GetVssConnection(message));
+            await jobServer.ConnectAsync(vssConnection);
+
+            var agentServer = HostContext.GetService<IAgentServer>();
+            await agentServer.ConnectAsync(vssConnection);
```

Comment:
I see why you need news up agent server, since the RM client is in agent server.
I would suggest move rm client into RMServer, sent connect in your Extension, otherwise it is weird the entire agent can access RM client.